evalml/automl/automl_search.py

```diff
@@ -289,24 +289,30 @@ def __init__(self,
         self.pipeline_parameters = pipeline_parameters if pipeline_parameters is not None else {}
         self.search_iteration_plot = None
         self._interrupted = False
+        self._frozen_pipeline_parameters = {}
+
+        parameters = copy.copy(self.pipeline_parameters)
+        if self.problem_configuration:
+            parameters.update({'pipeline': self.problem_configuration})
+            self._frozen_pipeline_parameters.update({'pipeline': self.problem_configuration})
```

Comment:
Here's the logic right now in `main`: 

We pass the pipeline parameters, noticeably *without* the drop-columns parameters, to `make_pipelines`. This means that there are no custom hyperparameters set for the drop-columns component. Then, we use IterativeAlgo to handle all the wrangling of this specific case. In main, in we move the creation of `allowed_pipelines` from L297 to L302, we would error out:
```
ValueError: Default parameters for components in pipeline Decision Tree Classifier w/ Imputer + Drop Columns Transformer not in the hyperparameter ranges: Point (['most_frequent', 'mean', ['index_col'], 'gini', 'auto', 6]) is not within the bounds of the space ([('most_frequent',), ('mean', 'median', 'most_frequent'), ('index_col',), ('gini', 'entropy'), ('auto', 'sqrt', 'log2'), (4, 10)]).
```
(https://github.com/alteryx/evalml/blob/main/evalml/automl/automl_search.py).

Here, I had passed in `parameters` (rather than a copy) to the custom hyperparameters, which caused the same error. It's still a little odd to me that this logic is dependent on us *not* passing extra information for the drop columns transformer here and then passing that information to the IterativeAlgorithm to handle later, and I was hoping to ease some of that, because if we have more components that take in lists as parameters, we'll have to hardcode more code... but also okay with not doing this for now if it convolutes the logic even further!

Reply:
EDIT: Actually, running some more tests, I think this might still be a good idea. Before, `allowed_pipelines` created classes. Now, we create instances. For time-series problems, we need to pass in the parameters. These parameters are found via `pipeline_parameters`. That's why this block of code was moved up, so that we can pass in the appropriate parameters to the time series pipeline instances. However, this logic that we currently have, `pipeline_parameters` also potentially contains smaller custom hyperparameter spaces, which will error out as parameters.

I noticed this when `test_automl_pipeline_params_kwargs` failed 😓 Open to filing an issue to somehow clean this up, or maybe this is the cleanup we need to be aware of as we restructure and rethink `IterativeAlgorithm`

Reply:
Ok, understood. If this looks good to you, I am on board. Yep, I hope we can revisit this after this PR and find a good way to clarify how we pass parameters to the automl algo.

Reply:
Yep, I'll take another pass at this before merging, and filed #2187 to revisit this!